idaes/core/unit_model.py

```diff
@@ -652,16 +652,26 @@ def initialize(blk, state_args=None, outlvl=idaeslog.NOTSET,
 
         # ---------------------------------------------------------------------
         # Solve unit
+
+        # if costing block exists, deactivate
+        if hasattr(blk, "costing"):
+            blk.costing.deactivate()
+
         with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
             results = opt.solve(blk, tee=slc.tee)
 
         init_log.info_high(
             "Initialization Step 2 {}.".format(idaeslog.condition(results))
         )
 
+        # if costing block exists, activate and initialize
+        if hasattr(blk, "costing"):
+            import idaes.core.util.unit_costing as costing
```

Comment:
Whilst this might work, I am not sure that I like the solution.

Firstly, it is generally considered good practice to put all imports at the beginning of the file.
Secondly, this isn't really addressing the problem. It just hides it by putting delaying the import until a point where costing has probably been imported by something else already (the thing causing the circular import).

I would much rather we work out what the actual issue is and fix that.

Reply:
"This approach doesn't contradict Python syntax, as the Python documentation says: "It is customary but not required to place all import statements at the beginning of a module (or script, for that matter)"."